modelcontextprotocol/tools/assets.py

```diff
@@ -60,24 +62,71 @@ def update_assets(
 
         # Create assets with updated values
         assets = []
+        # Maintain separate lists for readme updates to avoid confusion with other asset attribute updates.
+        # readme_update_assets: Readme assets to update.
+        # readme_update_parent_assets: Assets that were updated with readme.
+        readme_update_assets = []
+        readme_update_parent_assets = []
         index = 0
         for updatable_asset in updatable_assets:
             type_name = updatable_asset.type_name
+            qualified_name = updatable_asset.qualified_name
             asset_cls = getattr(
                 __import__("pyatlan.model.assets", fromlist=[type_name]), type_name
             )
             asset = asset_cls.updater(
                 qualified_name=updatable_asset.qualified_name,
                 name=updatable_asset.name,
             )
-            setattr(asset, attribute_name.value, attribute_values[index])
-            assets.append(asset)
+
+            # Special handling for README updates
+            if attribute_name == UpdatableAttribute.README:
+                response_readme_fetch = (
+                    FluentSearch()
+                    .select()
+                    .where(CompoundQuery.asset_type(asset_cls))
+                    .where(asset_cls.QUALIFIED_NAME.eq(qualified_name))
+                    .include_on_results(asset_cls.README)
+                    .include_on_relations(Readme.DESCRIPTION)
+                    .execute(client=client)
+                )
+
+                if first := response_readme_fetch.current_page():
+                    updated_content = attribute_values[index]
+                    updated_readme = Readme.creator(
+                        asset=first[0], content=updated_content
+                    )
+                    # Save the readme asset
+                    # Readme asset is created in the same request as the parent asset.
+                    readme_update_assets.append(updated_readme)
+                    # Add the parent/actual asset to the list of assets that were updated with readme.
+                    readme_update_parent_assets.append(asset)
```

Comment:
what if the readme does not exist? It should create one right?

Reply:
Resolved via : [bb457b3](https://github.com/atlanhq/agent-toolkit/pull/63/commits/bb457b340c3dcf9117a0e93391b2d97f43d0b5d5)